modules/ROOT/pages/subqueries/subqueries-in-transactions.adoc

```diff
@@ -550,3 +550,83 @@ RETURN largeLists
 1+d|Rows: 2
 
 |===
+
+[[composite-databases]]
+== Composite databases
+`CALL { ... } IN TRANSACTIONS` works on composite databases.
+Composite databases are virtual databases used for querying their constituents, so the examples above are generally
+valid for composite databases, but they must be slightly modified to accommodate the nature of composite databases.
+
+The following two examples shows how `CALL { ... } IN TRANSACTIONS` can be used on a composite database.
+Both examples are the equivalent of examples presented in the previous sections, but modified for composite databases.
+The first query uses 'friends.csv' from previous examples and creates the person nodes on all constituents.
+
+.friends.csv
+[source, csv, role="noheader" filename="friends.csv"]
+----
+1,Bill,26
+2,Max,27
+3,Anna,22
+4,Gladys,29
+5,Summer,24
+----
+
+.Query
+[source, cypher]
+----
+UNWIND graph.names() AS graphName
+LOAD CSV FROM 'file:///friends.csv' AS line
+CALL {
+  USE graph.byName( graphName )
+  WITH line
+  CREATE (:Person {name: line[1], age: toInteger(line[2])})
+} IN TRANSACTIONS
+----
+
+The second query removes all nodes from all constituents.
+
+.Query
+[source, cypher]
+----
+UNWIND graph.names() AS graphName
+CALL {
+  USE graph.byName( graphName )
+  MATCH (n)
+  RETURN id(n) AS id
+}
+CALL {
+  USE graph.byName( graphName )
+  WITH id
+  MATCH (n)
+  WHERE id(n) = id
+  DETACH DELETE n
+} IN TRANSACTIONS
+----
+
+The two examples demonstrate another useful feature of `CALL { ... } IN TRANSACTIONS`.
+Even though composite databases allow accessing multiple graphs in a single query,
+only one graph can be modified in one transaction. `CALL { ... } IN TRANSACTIONS`
+offers a way how to construct queries modifying multiple graphs.
+
+[[composite-databases-batching]]
+=== Batch size in composite databases
+Since queries on composite databases can interact with more than one graph, batching is slightly more complicated there.
+The batch size declared with  `IN TRANSACTIONS OF ... ROWS` represents an upper limit of the batch size. The real batch size
+depends on the evaluation of `USE` clause. Calls targeting different graphs can't be interleaved so if a `USE` clause
+evaluates to a different target than the current one, the batch is committed and a new one started.
+This is demonstrated in the following example. It assumes existence of two constituents 'remoteGraph1' and 'remoteGraph2'
+of composite database called 'composite'.
+
+.Query
+[source, cypher]
+----
+WITH ['composite.remoteGraph1', 'composite.remoteGraph2'] AS graphs
+UNWIND [0, 0, 1, 1, 1, 1, 0, 0] AS i
+WITH graphs[i] AS g
+CALL {
+  USE graph.byName( g )
+  CREATE ()
+} IN TRANSACTIONS OF 3 ROW
+----
+The declared batch size is 3, but the evaluation of USE clause will cause the batch size to be 2 on 'composite.remoteGraph1',
+followed by 3 on 'composite.remoteGraph2', followed by 1 on 'composite.remoteGraph2' and finally 2 on 'composite.remoteGraph1'.
```

Comment:
Do we want to comment here that if there's an error during execution, the part of the error message that says "(Transactions committed: 0)" should be ignored/is not accurate

Reply:
Good point. I guess we mention it as a current limitation.

Reply:
Where do current limitations live?

Reply:
Currently nowhere :) .

Maybe let's add it as a note?
[NOTE]
====
There is a currently a known issue.
When an error occurs during CALL In Transaction processing, the error message includes the information about how many transaction have been committed.
The number of reported committed transaction is always 0 in error messages on composite databases.
====

Reply:
@htsedebenham @rsill-neo4j what do you think?

Reply:
```
[NOTE]
====
There is a currently a known issue.
When an error occurs during CALL In Transaction processing, the error message includes information about how many transactions have been committed.
Errors on composite databases always report "(Transactions committed: 0)".
====
```

Reply:
May as well let SEO try and direct here 😅

Reply:
What's the ETA to fixing this? Is there a card to which removing this note has been noted down, so that that admonition doesn't say longer than necessary? Or is the issue planned to stay longer term?

Reply:
It will stay like that a couple of releases. Runtime team will start working on a better implementation soon that will not have this limitation. I will ask them to include a task to remove this note in the related Trello cards.